exchanges/btcmarkets/btcmarkets.go

```diff
@@ -53,316 +84,555 @@ type BTCMarkets struct {
 
 // GetMarkets returns the BTCMarkets instruments
 func (b *BTCMarkets) GetMarkets() ([]Market, error) {
-	type marketsResp struct {
-		Response
-		Markets []Market `json:"markets"`
-	}
-
-	var resp marketsResp
-	path := fmt.Sprintf("%s/v2/market/active", b.API.Endpoints.URL)
-
-	err := b.SendHTTPRequest(path, &resp)
-	if err != nil {
-		return nil, err
-	}
-
-	if !resp.Success {
-		return nil, fmt.Errorf("%s unable to get markets: %s", b.Name, resp.ErrorMessage)
-	}
-
-	return resp.Markets, nil
+	var resp []Market
+	return resp, b.SendHTTPRequest(btcMarketsUnauthPath, &resp)
 }
 
 // GetTicker returns a ticker
 // symbol - example "btc" or "ltc"
-func (b *BTCMarkets) GetTicker(firstPair, secondPair string) (Ticker, error) {
-	tick := Ticker{}
-	path := fmt.Sprintf("%s/market/%s/%s/tick",
-		b.API.Endpoints.URL,
-		strings.ToUpper(firstPair),
-		strings.ToUpper(secondPair))
-
-	return tick, b.SendHTTPRequest(path, &tick)
-}
-
-// GetOrderbook returns current orderbook
-// symbol - example "btc" or "ltc"
-func (b *BTCMarkets) GetOrderbook(firstPair, secondPair string) (Orderbook, error) {
-	orderbook := Orderbook{}
-	path := fmt.Sprintf("%s/market/%s/%s/orderbook",
-		b.API.Endpoints.URL,
-		strings.ToUpper(firstPair),
-		strings.ToUpper(secondPair))
-
-	return orderbook, b.SendHTTPRequest(path, &orderbook)
+func (b *BTCMarkets) GetTicker(marketID string) (Ticker, error) {
+	var tick Ticker
+	return tick, b.SendHTTPRequest(btcMarketsUnauthPath+marketID+btcMarketsGetTicker, &tick)
 }
 
 // GetTrades returns executed trades on the exchange
-// symbol - example "btc" or "ltc"
-// values - optional paramater "since" example values.Set(since, "59868345231")
-func (b *BTCMarkets) GetTrades(firstPair, secondPair string, values url.Values) ([]Trade, error) {
+func (b *BTCMarkets) GetTrades(marketID string, before, after, limit int64) ([]Trade, error) {
 	var trades []Trade
-	path := common.EncodeURLValues(fmt.Sprintf("%s/market/%s/%s/trades",
-		b.API.Endpoints.URL, strings.ToUpper(firstPair),
-		strings.ToUpper(secondPair)), values)
-
-	return trades, b.SendHTTPRequest(path, &trades)
+	params := url.Values{}
+	if before != 0 {
```

Comment:
Please add a check here to see if before and after are both set like what you've done in the other funcs